examples/history-read/history-read.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2018-2019 opcua authors. All rights reserved.
+// Use of this source code is governed by a MIT-style license that can be
+// found in the LICENSE file.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/gopcua/opcua"
+	"github.com/gopcua/opcua/debug"
+	"github.com/gopcua/opcua/id"
+	"github.com/gopcua/opcua/ua"
+)
+
+func main() {
+	endpoint := flag.String("endpoint", "opc.tcp://localhost:4840", "OPC UA Endpoint URL")
+	nodeID := flag.String("node", "", "NodeID to read")
+	flag.BoolVar(&debug.Enable, "debug", false, "enable debug logging")
+	flag.Parse()
+	log.SetFlags(0)
+
+	c := opcua.NewClient(*endpoint)
+	if err := c.Connect(); err != nil {
+		log.Fatal(err)
+	}
+	defer c.Close()
+
+	node, err := ua.ParseNodeID(*nodeID)
+	if err != nil {
+		log.Fatalf("invalid node id: %v", err)
+	}
+
+	// HistoryRead with ContinuationPoint use
+	nodesToRequest := make([]*ua.HistoryReadValueID, 1)
+	nodesToRequest[0] = &ua.HistoryReadValueID{
+		NodeID:       node,
+		DataEncoding: &ua.QualifiedName{},
+	}
+
+	for {
+		if len(nodesToRequest) < 1 {
+			break
+		}
+
+		// For ContinuationPoint usage
+		// Get current nodes
+		nodes := nodesToRequest
+		// Reset old nodes
+		nodesToRequest = make([]*ua.HistoryReadValueID, 0)
+
+		// Part 4, 5.10.3 HistoryRead
+		req := &ua.HistoryReadRequest{
+			TimestampsToReturn:        ua.TimestampsToReturnSource,
+			ReleaseContinuationPoints: false,
```

Comment:
You can omit all fields set to their zero values, e.g. `false` and `0` for `bool` and `int` fields.